### DISCUSSION TITLE
Monero Ocean Pool Showing zero profit

### DISCUSSION THREAD

**Discussion by m-c-hizzle** [RainbowMiner/RainbowMiner#1679] [Q&A] [answered]
I have rainbow miner set to only use monero ocean pool and to only use the gpu mineable algorithms - eth, cngpu, c29s, c29b, kawpow. ETH should obviously be #1 to mine at this point but there is some kind of pricing error. In the best pools tab, ETH displays 0 btc/gh/day. Also there is a conflict related to this where the "miners" tab will also display zero profit and revenue for ETH.
The "active mienrs" tab doesnt even show the option for eth and has also removed c29s. At this point, with 2 algos removed, rainbow miner is not using the most profitable algorithm.
![best pools](https://user-images.githubusercontent.com/85482477/120982227-4fba5780-c72d-11eb-93dc-b50ff1b40a45.JPG)
![miners](https://user-images.githubusercontent.com/85482477/120982359-7f695f80-c72d-11eb-8035-ef6570904738.JPG)
![active miners](https://user-images.githubusercontent.com/85482477/120982364-81332300-c72d-11eb-986f-473f78882120.JPG)

**Comment by RainbowMiner** [comment, marked as answer]
It's a feature of RainbowMiner:
it takes into account the average time to find a block and the time that has past, since the last block has been found.

Ethash gets pretty low hashrate on MoneroOcean, so that leads to a long time to find (blocks can be monitored on the "Blocks" page at moneroocean.stream). For profit switching, it is in most cases a good idea to avoid mining to those blocks, since the revenue drops significantly, if the ttf is much longer than the world average ttf. 

But - you can easily deactivate the feature in pools.config.txt, "MoneroOcean" section. Just set the following two parameters to "0":

```
    "MaxAllowedLuck": "0",
    "MaxTimeSinceLastBlock": "0",
```

That's from the documentation:
- **MaxAllowedLuck** = if luck (= time_since_last_block / time_to_find_one_block) is greater than that number, increase the penalty up to 100% (0=disable, inactive for solo pools) [default=3]
- **MaxTimeSinceLastBlock** = if time_since_last_block is greater than that number, increase the penalty up to 100% (0=disable, time-units allowed in input, inactive for solo pools) [default=12h]

**Comment by RainbowMiner** [comment]
> has also removed c29s.

Please check
- the "Watchdog" on http://localhost:4000, it's possible, that C29s is disabled due to miner fail (and/or pool fails to report back any found shares)
- the "Benchmarks" on http://localhost:4000, it's possible, that Cuckaroo29s failed to benchmark for some reason.

**Comment by m-c-hizzle** [comment]
Under benchmarks, neither c29s or c29b appear now, but they did 8 hours ago. Also c29s appears in best pools but c29b does not, this is reversed from when I asked the first question. They do not appear in the active miners tab either.
They 100% did benchmark and were showing up when I setup rainbow miner, also there is nothing to do with the c29 algorithms in the watchdog settings, but I did reset that anyway.

**Comment by RainbowMiner** [comment]
Ok, there is another feature to optimize revenue - by default, RainbowMiner skips algorithms at pools, if there is no worker/no hashrate for it.

To disable this, again edit pools.config.txt, section "MoneroOcean" and set the following parameter to "1":
```
    "AllowZero": "1",
```

**Comment by haley0918** [comment]
I have set up RainbowMiner and also use MoneroOcean pool.
C29 is listed on best pools tab but missing from miners and benchmark.
AllowZero is enabled.
